src/commands/deploy.ts

```diff
@@ -46,6 +47,9 @@ export async function deploy(tree: AzureTreeDataProvider, outputChannel: vscode.
 }
 
 async function getJavaFolderPath(outputChannel: vscode.OutputChannel, basePath: string, ui: IUserInterface): Promise<string> {
+    if (!(await mavenUtils.isMavenInstalled(basePath))) {
+        throw new Error(localize('azFunc.mvnNotFound', 'Failed to find "maven" on path.'));
```

Comment:
Don't copy code like this - you could just refactor `isMavenInstalled` to `validateMavenInstalled` and have it throw the error